src/Compilers/CSharp/Test/Symbol/Compilation/GetSemanticInfoTests.cs

```diff
@@ -6048,5 +6048,62 @@ class B<T, U, U>
             symbol = model.GetDeclaredSymbol(typeParameters[typeParameters.Length - 2]);
             Assert.True(symbol.IsReferenceType);
         }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/75570")]
+        public void GetConstructedReducedFrom_WithNullableInferredType()
+        {
+            var source = """
+                #nullable enable
+
+                using System.Collections.Generic;
+                using System.Diagnostics.CodeAnalysis;
+                using System.Linq;
+
+                class C1 { }
+
+                class C2 { }
+
+                class D
+                {
+                    public static void ConvertAll(IEnumerable<C1> values)
+                    {
+                        var v = values.Select(Convert);
+                    }
+
+                    [return: NotNullIfNotNull(nameof(value))]
+                    private static C2? Convert(C1? value) => value is null ? null : new C2();
+                }
+                """ + NotNullIfNotNullAttributeDefinition;
+
+            var comp = CreateCompilationWithMscorlib40AndSystemCore(source).VerifyDiagnostics();
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var memberAccess = tree.GetRoot().DescendantNodes().OfType<MemberAccessExpressionSyntax>().First();
+            var constructedMethodSymbol = (IMethodSymbol)model.GetSymbolInfo(memberAccess).Symbol;
+
+            // Ensure that both the bound reduced method symbol, and the constructed method symbol 
+            // it was reduced from both have the right nullable annotations for the type parameters.
+            Assert.True(constructedMethodSymbol is
```

Comment:
It may be a bit simpler and easier to evaluate a difference, if a `ToTestDisplayString` including the nullability were used. However, no need to make a change, if you prefer this way.

Reply:
I personally like hitting the symbols directly :)